extensions/wikia/AdEngine/js/slot/scrollHandler.js

```diff
@@ -1,67 +1,78 @@
 /*global define*/
 define('ext.wikia.adEngine.slot.scrollHandler', [
-    'ext.wikia.adEngine.adContext',
-    'ext.wikia.adEngine.adHelper',
-    'wikia.log',
-    'wikia.document',
-    'wikia.window'
+	'ext.wikia.adEngine.adContext',
+	'ext.wikia.adEngine.adHelper',
+	'wikia.log',
+	'wikia.document',
+	'wikia.window'
 ], function (adContext, adHelper, log, doc, win) {
-    'use strict';
+	'use strict';
 
-    var logGroup = 'ext.wikia.adEngine.slot.scrollHandler',
-        isRefreshed = {
-            PREFOOTER_LEFT_BOXAD: false,
-            PREFOOTER_RIGHT_BOXAD: false
-        },
-        reloadedView = {
-            PREFOOTER_LEFT_BOXAD: 0,
-            PREFOOTER_RIGHT_BOXAD: 0
-        };
+	var logGroup = 'ext.wikia.adEngine.slot.scrollHandler',
+		isRefreshed = {},
+		reloadedView = {},
+		context = adContext.getContext(),
+		config = context.opts.scrollHandlerConfig ||
+			{
+				PREFOOTER_LEFT_BOXAD: {reloadedViewMax: 1},
+				PREFOOTER_RIGHT_BOXAD: {reloadedViewMax: 3}
+			};
 
-    function init() {
-        if (adContext.getContext().opts.enableScrollHandler) {
-            win.addEventListener('scroll', adHelper.throttle(function () {
-                log('Scroll event listener has been added', 'debug', logGroup);
-                for (var slotName in isRefreshed) {
-                    if (isRefreshed.hasOwnProperty(slotName)) {
-                        refreshSlot(slotName);
-                    }
-                }
-            }));
-        }
-    }
+	function init() {
+		if (context.opts.enableScrollHandler) {
+			for (var slotName in config) {
```

Comment:
To make it less indentation-hellish you could extract the logic to two other functions, so the `init` function could be really small and clear:
```
function init() {
   if (context.opts.enableScrollHandler) {
      prepareSettings(config);
      registerSlotEvents();
   }
}
```